src/material-luxon-adapter/adapter/luxon-date-adapter.ts

```diff
@@ -0,0 +1,254 @@
+    @Optional() @Inject(MAT_LUXON_DATE_ADAPTER_OPTIONS)
+    options?: MatLuxonDateAdapterOptions) {
+
+    super();
+    this._useUTC = options ? !!options.useUtc : false;
+    this.setLocale(dateLocale || DateTime.local().locale);
+  }
+
+  getYear(date: DateTime): number {
+    return date.year;
+  }
+
+  getMonth(date: DateTime): number {
+    // Luxon works with 1-indexed months whereas our code expects 0-indexed.
+    return date.month - 1;
+  }
+
+  getDate(date: DateTime): number {
+    return date.day;
+  }
+
+  getDayOfWeek(date: DateTime): number {
+    return date.weekday;
+  }
+
+  getMonthNames(style: 'long' | 'short' | 'narrow'): string[] {
+    return Info.months(style, {locale: this.locale});
+  }
+
+  getDateNames(): string[] {
+    // At the time of writing, Luxon doesn't offer similar
+    // functionality so we have to fall back to the Intl API.
+    const dtf = new Intl.DateTimeFormat(this.locale, {day: 'numeric', timeZone: 'utc'});
+
+    return range(31, i => {
+      // Format a UTC date in order to avoid DST issues.
+      const date = DateTime.utc(2017, 1, i + 1).toJSDate();
+
+      // Strip the directionality characters from the formatted date.
+      return dtf.format(date).replace(/[\u200e\u200f]/g, '');
+    });
+  }
+
+  getDayOfWeekNames(style: 'long' | 'short' | 'narrow'): string[] {
+    // Note that we shift the array once, because Luxon returns Monday as the
+    // first day of the week, whereas our logic assumes that it's Sunday. See:
+    // https://moment.github.io/luxon/docs/class/src/info.js~Info.html#static-method-weekdays
+    const days = Info.weekdays(style, {locale: this.locale});
+    days.unshift(days.pop()!);
+    return days;
+  }
+
+  getYearName(date: DateTime): string {
+    return date.toFormat('yyyy');
+  }
+
+  getFirstDayOfWeek(): number {
+    // Luxon doesn't have support for getting the first day of the week.
+    return 0;
+  }
+
+  getNumDaysInMonth(date: DateTime): number {
+    return date.daysInMonth;
+  }
+
+  clone(date: DateTime): DateTime {
+    return DateTime.fromObject(date.toObject());
+  }
+
+  createDate(year: number, month: number, date: number): DateTime {
+    if (month < 0 || month > 11) {
+      throw Error(`Invalid month index "${month}". Month index has to be between 0 and 11.`);
+    }
+
+    if (date < 1) {
+      throw Error(`Invalid date "${date}". Date has to be greater than 0.`);
+    }
+
+    // Luxon uses 1-indexed months so we need to add one to the month.
+    const result =
+        this._useUTC ? DateTime.utc(year, month + 1, date) : DateTime.local(year, month + 1, date);
+
+    if (!this.isValid(result)) {
+      throw Error(`Invalid date "${date}". Reason: "${result.invalidReason}".`);
+    }
+
+    return result.setLocale(this.locale);
+  }
+
+  today(): DateTime {
+    return (this._useUTC ? DateTime.utc() : DateTime.local()).setLocale(this.locale);
+  }
+
+  parse(value: any, parseFormat: string | string[]): DateTime | null {
```

Comment:
Should we tighten the type of `value` to what we actually support in the function? Or potentially even `unknown` since we're checking the type anyway.

Reply:
It's an `any`, because the `DateAdapter` interface has it as an `any` too.

Reply:
Yeah, I think it makes sense for the abstract adapter to be `any`, but IIRC subclasses are able to narrow signatures, right? (Non-blocking)